python/vulns_per_assets/export_asset_vulns.py

```diff
@@ -0,0 +1,241 @@
+# Does an asset export and for each asset, collects the vulnerability information.
+# A gzip file in the form of assets_xxxx.gz and a unzip file assets_xxxx are left
+# around.
+#
+# The script has one optional parameter, the search ID.  If specified, it is assumed
+# that you know the search ID from a previous search.
+
+import os
+import sys
+import time
+import requests
+import json
+import gzip
+import shutil
+import math
+
+# Invoke the data_exports API to request an asset export.
+def request_asset_exports(api_key, base_url):
+    url = base_url + "/data_exports"
+    headers = {'Accept': 'application/json',
+               'Content-Type': 'application/json; charset=utf-8',
+               'X-Risk-Token': api_key}
+
+    filter_params = {
+        'status' : ['active'],
+        #'records_updated_since' : 'now-01d',
+        'export_settings': {
+            'format': 'jsonl',
+            'model': 'asset'
+        }
+    }
+    
+    try:
+        response = requests.post(url, headers=headers, data=json.dumps(filter_params))
+    except Exception as exp:
+        print("Assets Data Exports Error: " + exp.__str__())
+        exit(1)
+    
+    resp = response.json()
+    search_id = str(resp['search_id'])
+    asset_count = resp['record_count']
+    print(f"Search ID: {id}")
+    print(f"Asset count: {asset_count}")
+    return (search_id, asset_count)
+
+def get_export_status(api_key, base_url, search_id):
+    check_status_url = base_url + "/data_exports/status?search_id=" + search_id
+    headers = {'Accept': 'application/json',
+               'Content-Type': 'application/json; charset=utf-8',
+               'X-Risk-Token': api_key}
+
+    try:
+        response = requests.get(check_status_url, headers=headers)
+    except Exception as exp:
+        print("Get Export Status Error: " + exp.__str__())
```

Comment:
This another instance of an odd usage for the str overload.

Reply:
Fixed.